src/Testcontainers/Containers/DockerContainer.cs

```diff
@@ -405,19 +405,29 @@ await this.client.AttachAsync(this.container.ID, this.configuration.OutputConsum
       await this.client.StartAsync(this.container.ID, ct)
         .ConfigureAwait(false);
 
-      this.container = await this.client.InspectContainer(this.container.ID, ct)
+      // Do not use a too small frequency. The Docker endpoint cancels too many concurrent operations (requests).
+      var frequency = (int)TimeSpan.FromSeconds(1).TotalMilliseconds;
+
+      var timeout = (int)TimeSpan.FromSeconds(15).TotalMilliseconds;
+
+      async Task<bool> CheckPortBindings()
+      {
+        this.container = await this.client.InspectContainer(this.container.ID, ct)
+          .ConfigureAwait(false);
+
+        var boundPorts = this.container.NetworkSettings.Ports.Values.Where(portBindings => portBindings != null).SelectMany(portBinding => portBinding).Count(portBinding => !string.IsNullOrEmpty(portBinding.HostPort));
+        return this.configuration.PortBindings == null || /* IPv4 or IPv6 */ this.configuration.PortBindings.Count == boundPorts || /* IPv4 and IPv6 */ 2 * this.configuration.PortBindings.Count == boundPorts;
+      }
```

Comment:
@cristianrgreco @eddumelendez @kiview @mdelapenya  I am interessted in your opinion on how you handle a particular situation in other language implementations. When using the Docker.DotNet `InspectContainer(string, CancellationToken)` member, it is possible to receive a response that does not immediately contain all port bindings. In most cases, this is not an issue. Soon as the wait strategy indicate readiness, we receive an instance that contains all information.

However, there are two cases where it may be useful to wait until the instance received all port bindings:

- The user did not set a wait strategy (although this is probably not a critical one).
- Custom code is being run during startup, as it is the case with [Couchbase](https://github.com/testcontainers/testcontainers-java/blob/74e8954f55401aa4fea64aed8d63e4cc3e258316/modules/couchbase/src/main/java/org/testcontainers/couchbase/CouchbaseContainer.java#L336-L350).

Reply:
As part of the container startup code in Java, we will wait until the inspect command returns all mapped ports:
https://github.com/testcontainers/testcontainers-java/blob/6fe954ab940ccb674d85aca543fe4c979e8932f0/core/src/main/java/org/testcontainers/containers/GenericContainer.java#L450

Only afterwards the callback method of extending containers is called. So the callback methods can be sure to already get a consistent response regarding the ports.